gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -122,6 +126,9 @@ public class GoogleCloudStorageImpl implements GoogleCloudStorage {
           .setEnsureEmptyObjectsMetadataMatch(false)
           .build();
 
+  private static Cache<String, Boolean> cache =
```

Comment:
Note that we need to check if the bucket is HN enabled before we use rename operation. We are caching this information, rather than calling this before each rename.

Reply:
I understand that, I meant to say we can improve of variable name.

Reply:
fair point.